dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala

```diff
@@ -158,12 +164,19 @@ class AkkaHttpServiceGateway(lifecycle: ApplicationLifecycle, config: ServiceGat
   }
 
   private val bindingFuture = Http().bindAndHandle(handler, config.host, config.port)
-  lifecycle.addStopHook(() => {
-    for {
-      binding <- bindingFuture
-      unbind <- binding.unbind()
-    } yield unbind
-  })
+
+  coordinatedShutdown.addTask(
+    CoordinatedShutdown.PhaseServiceUnbind,
+    "unbind-akka-http-service-gateway"
+  ) {
+      () =>
+        {
+          (for {
+            binding <- bindingFuture
+            unbind <- binding.unbind()
+          } yield unbind).map(_ => Done)
```

Comment:
I had to squint at this a bit to see where the parens match, could it be simplified to this?

```scala
for {
  binding <- bindingFuture
  _       <- binding.unbind()
} yield Done
```

or else this?

```scala
bindingFuture.flatMap(_.unbind).map(_ => Done)
```